gammapy/scripts/jupyter.py

```diff
@@ -0,0 +1,166 @@
+    log.error("Jupyter package not presen, 'jupyter' option is not available.")
+    sys.exit()
+
+
+@click.command(name='execute')
+@click.pass_context
+def cli_jupyter_execute(ctx):
+    """Execute Jupyter notebooks."""
+
+    for path in ctx.obj['paths']:
+        run_notebook(path)
+
+
+def run_notebook(path, loglevel=20):
+    """Execute a Jupyter notebook."""
+
+    try:
+        t = time.time()
+        subprocess.call(
+            "jupyter nbconvert "
+            "--allow-errors "
+            "--log-level={} "
+            "--ExecutePreprocessor.timeout=None "
+            "--ExecutePreprocessor.kernel_name=python3 "
+            "--to notebook "
+            "--inplace "
+            "--execute '{}'".format(loglevel, path),
+            shell=True)
+        t = (time.time() - t) / 60
+        log.info('   ... Executing duration: {:.2f} mn'.format(t))
+    except Exception as ex:
+        log.error('Error executing file {}'.format(str(path)))
+        log.error(ex)
+
+
+@click.command(name='stripout')
+@click.pass_context
+def cli_jupyter_stripout(ctx):
+    """Strip output cells."""
+
+    for path in ctx.obj['paths']:
+        rawnb = nbformat.read(str(path), as_version=nbformat.NO_CONVERT)
+
+        for cell in rawnb.cells:
+            if cell['cell_type'] == 'code':
+                cell['outputs'] = []
+
+        nbformat.write(rawnb, str(path))
+        log.info('Jupyter notebook {} stripped out.'.format(str(path)))
+
+
+@click.command(name='black')
+@click.pass_context
+def cli_jupyter_black(ctx):
+    """Format code cells with black."""
+    try:
+        import black
+    except ImportError:
+        log.error("Black package not present, 'black' option is not available.")
+        sys.exit()
+
+    for path in ctx.obj['paths']:
+        rawnb = nbformat.read(str(path), as_version=nbformat.NO_CONVERT)
+        blacknb = BlackNotebook(rawnb)
+        blacknb.blackformat()
+        rawnb = blacknb.rawnb
+        nbformat.write(rawnb, str(path))
+        log.info('Jupyter notebook {} blacked.'.format(str(path)))
+
+
+class BlackNotebook:
+    """Manage the process of black formatting."""
+
+    MAGIC_TAG = "###-MAGIC TAG-"
+
+    def __init__(self, rawnb):
+
+        self.rawnb = rawnb
+
+    def blackformat(self):
+        """Format code cells."""
+        from black import format_str
+        for cell in self.rawnb.cells:
+            fmt = cell['source']
+            if cell['cell_type'] == 'code':
+                try:
+                    fmt = '\n'.join(self.tag_magics(fmt))
+                    has_semicolon = fmt.endswith(';')
+                    fmt = format_str(src_contents=fmt,
+                                     line_length=79).rstrip()
+                    if has_semicolon:
+                        fmt += ';'
+                except Exception as ex:
+                    logging.info(ex)
+                fmt = fmt.replace(self.MAGIC_TAG, "")
+            cell['source'] = fmt
+
+    def tag_magics(self, cellcode):
```

Comment:
Make this a `@staticmethod` since self isn't accessed?

Reply:
There is `self.MAGIC_TAG`
How about moving `self.MAGIC_TAG` out of the function and have it s a global var in `jupyter.py`? At this moment we could have @staticmethod

Reply:
No, then just leave as-is. It's perfectly fine, I just didn't see the access to `self` in this method.

Reply:
Better to have everything on this one class. Then it's easier to maintain and others can copy & paste it.

Actually, I think a separate `blacknb` or `black-notebook` Python package that does this would find a userbase; or we could file an issue in one of the Jupyter repos asking if they think it's appropriate to be included there directly.